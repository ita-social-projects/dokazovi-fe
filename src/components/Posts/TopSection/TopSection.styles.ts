import { makeStyles, Theme } from '@material-ui/core';

export const useStyles = makeStyles(
  (theme: Theme) => ({
    root: {
      minHeight: '100px',
      display: 'flex',
      alignItems: 'center',
      marginBottom: theme.spacing(5),
    },
    avatar: {
      height: 150,
      width: 150,
      borderRadius: '50%',
      filter: 'grayscale(100%)',
      marginRight: theme.spacing(4),
    },
    authorName: {
      fontSize: '24px',
      lineHeight: '1.17',
    },
    authorBio: {
      fontSize: '15px',
      lineHeight: '1.47',
    },
  }),
  { name: 'PostView' },
);
