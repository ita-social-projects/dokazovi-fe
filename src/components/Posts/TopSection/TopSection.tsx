import { Box, CardMedia, Typography } from '@material-ui/core';
import React from 'react';
import { Link } from 'react-router-dom';
import { useStyles } from './TopSection.styles';

interface ITopSection {
  author: {
    avatar?: string;
    firstName: string;
    id: number;
    lastName: string;
    bio?: string;
  };
}

export default function TopSection({ author }: ITopSection): JSX.Element {
  const classes = useStyles();
  const { id, avatar = '', firstName, lastName, bio } = author;
  const authorFullName = `${firstName} ${lastName}`;
  return (
    <Box className={classes.root}>
      <Link to={`/experts/${id}`}>
        <CardMedia
          className={classes.avatar}
          image={avatar} // paste default avatar if not present
          title={authorFullName}
          component="div"
        />
      </Link>
      <Box>
        <Link to={`/experts/${id}`}>
          <Typography variant="h3" className={classes.authorName}>
            {authorFullName}
          </Typography>
        </Link>
        <Typography color="textSecondary" className={classes.authorBio}>
          {bio}
        </Typography>
      </Box>
    </Box>
  );
}
