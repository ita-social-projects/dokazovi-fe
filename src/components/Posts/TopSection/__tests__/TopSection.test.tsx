import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import TopSection, { ITopSection } from '../TopSection';

describe('TopSection tests', () => {
  const topSectionMocks: ITopSection = {
    author: {
      avatar: 'https://i.pravatar.cc/300?img=20',
      firstName: 'Єва',
      id: 35,
      lastName: 'Євенко',
      bio:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    },
  };

  it('should TopSection Component render correctly', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <TopSection author={topSectionMocks.author} />
      </BrowserRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should main blocks be in the TopSection Component', () => {
    render(
      <BrowserRouter>
        <TopSection author={topSectionMocks.author} />
      </BrowserRouter>,
    );

    expect(screen.getByTestId('avatarSection')).toBeInTheDocument();
    expect(screen.getByTestId('infoAuthor')).toBeInTheDocument();
  });

  it('should component be without avatar', () => {
    delete topSectionMocks.author.avatar;

    const { asFragment } = render(
      <BrowserRouter>
        <TopSection author={topSectionMocks.author} />
      </BrowserRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
