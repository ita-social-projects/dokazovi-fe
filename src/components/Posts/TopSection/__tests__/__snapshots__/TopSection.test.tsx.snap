// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSection tests should TopSection Component render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-6 PostView-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-7 PostView-avatarSection-2"
      data-testid="avatarSection"
    >
      <a
        href="/experts/35"
      >
        <div
          class="MuiCardMedia-root PostView-avatar-3"
          style="background-image: url(https://i.pravatar.cc/300?img=20);"
          title="Єва Євенко"
        />
      </a>
    </div>
    <div
      class="MuiBox-root MuiBox-root-8"
      data-testid="infoAuthor"
    >
      <a
        href="/experts/35"
      >
        <h3
          class="MuiTypography-root PostView-authorName-4 MuiTypography-h3"
        >
          Єва Євенко
        </h3>
      </a>
      <p
        class="MuiTypography-root PostView-authorBio-5 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TopSection tests should component be without avatar 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-22 PostView-root-17"
  >
    <div
      class="MuiBox-root MuiBox-root-23 PostView-avatarSection-18"
      data-testid="avatarSection"
    >
      <a
        href="/experts/35"
      >
        <div
          class="MuiCardMedia-root PostView-avatar-19"
          title="Єва Євенко"
        />
      </a>
    </div>
    <div
      class="MuiBox-root MuiBox-root-24"
      data-testid="infoAuthor"
    >
      <a
        href="/experts/35"
      >
        <h3
          class="MuiTypography-root PostView-authorName-20 MuiTypography-h3"
        >
          Єва Євенко
        </h3>
      </a>
      <p
        class="MuiTypography-root PostView-authorBio-21 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
